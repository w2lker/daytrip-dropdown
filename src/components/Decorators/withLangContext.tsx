import React from "react";

function withLangContext <Props, Lang>(Child: React.ComponentType<Props>, Context: React.Context<Lang>, lang: Lang):React.FC<Props> {
  const ContextProvider: React.FC<Props> = (props) => {
    return (
      <Context.Provider value={lang}>
        <Child {...props}/>
      </Context.Provider>
      )
  };
  ContextProvider.displayName = `WithLangContext(${Child.displayName}, ${Context.displayName})`;
  return ContextProvider;
}

export default withLangContext;
