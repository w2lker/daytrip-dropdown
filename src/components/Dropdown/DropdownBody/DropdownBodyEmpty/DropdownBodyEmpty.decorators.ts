import withStyles from "react-jss";

import DropdownBodyEmpty from './DropdownBodyEmpty';
import dropdownBodyEmptyStyles from './DropdownBodyEmpty.styles';

const styledComponent = withStyles(dropdownBodyEmptyStyles)(DropdownBodyEmpty);

export default styledComponent;
