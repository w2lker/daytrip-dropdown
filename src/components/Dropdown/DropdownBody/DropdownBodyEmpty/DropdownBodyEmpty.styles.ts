const dropdownBodyEmptyStyles = {
  wrapper: {
    position: 'absolute',
    top: '50%',
    width: '100%',
    transform: 'translateY(-50%)',
    textAlign: 'center',
  },
  icon: {
    margin: '0 0 20px',
    color: '#B2C0C7',
    '& svg': {
      display: 'block',
      margin: '0 auto',
      width: 60,
    }
  },
  title: {
    fontWeight: 500,
    fontSize: 14,
    color: '#546F7B',
    paddingBottom: 10,
    margin: 0
  },
  description: {
    padding: '0 30px',
    fontSize: 14,
    color: '#546F7B',
    lineHeight: '24px',
    margin: 0
  },
};

export default dropdownBodyEmptyStyles;
