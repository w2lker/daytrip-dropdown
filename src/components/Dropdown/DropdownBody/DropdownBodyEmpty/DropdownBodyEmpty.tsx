import React, {useContext} from 'react';
import {WithStyles} from "react-jss";

import dropdownBodyEmptyStyles from "./DropdownBodyEmpty.styles";
import DropdownLang from "../../DropdownRoot/DropdownRoot.lang";

import {emptySearch} from "./DropdownBodyEmpty.svg";
import lang from "../../../../const/lang";

export interface IDropdownBodyEmptyProps extends WithStyles<typeof dropdownBodyEmptyStyles>{
  isEmpty: boolean;
  isFilteredEmpty: boolean;
}

const DropdownBodyEmpty: React.FC<IDropdownBodyEmptyProps> = (props) => {
  const {isEmpty, isFilteredEmpty, classes} = props;
  // Fallback is case context is not connected
  const contextLang = useContext(DropdownLang) as typeof lang.dropdown;
  const emptyTextItem = { title: '', description: '' };
  const defaultTexts = contextLang && contextLang.emptyStates ? contextLang.emptyStates : { noOptionsProvided: emptyTextItem, noFilteredOptions: emptyTextItem };

  if( !(isEmpty || isFilteredEmpty) ) return null;
  const texts = isEmpty ? defaultTexts.noOptionsProvided : defaultTexts.noFilteredOptions;

  return (
    <div className={classes.wrapper}>
      <div className={classes.icon}>
        {emptySearch()}
      </div>
      <h5 className={classes.title}>{texts.title}</h5>
      <p className={classes.description}>{texts.description}</p>
    </div>
  );
};

export default DropdownBodyEmpty;
