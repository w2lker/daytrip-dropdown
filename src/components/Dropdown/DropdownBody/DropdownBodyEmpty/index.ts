import DropdownBodyEmpty from './DropdownBodyEmpty.decorators';

export default DropdownBodyEmpty;
