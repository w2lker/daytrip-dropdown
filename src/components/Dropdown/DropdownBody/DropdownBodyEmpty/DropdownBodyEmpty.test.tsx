import React from 'react';

import { mount } from 'enzyme';

import DropdownBodyEmpty, { IDropdownBodyEmptyProps } from './DropdownBodyEmpty';
import DropdownLang from "../../DropdownRoot/DropdownRoot.lang";
import DropdownBodyEmptyDecorated from './DropdownBodyEmpty.decorators';

import dropdownBodyEmptyStyles from './DropdownBodyEmpty.styles';
import withLangContext from "../../../Decorators/withLangContext";

const setup = () => {
  const classes = {
    wrapper: 'dropdownBodyEmpty-wrapper',
    icon: 'dropdownBodyEmpty-icon',
    title: 'dropdownBodyEmpty-title',
    description: 'dropdownBodyEmpty-description',
  };
  const classesKeys = Object.keys(classes);
  const emptyProps: IDropdownBodyEmptyProps = {
    isEmpty: false,
    isFilteredEmpty: false,
    classes,
  };

  const sampleTexts = {
    emptyStates: {
      noOptionsProvided: {
        title: 'noOptions title',
        description: 'noOptions description',
      },
      noFilteredOptions: {
        title: 'noFiltered title',
        description: 'noFiltered description',
      },
    },
  };

  const ComponentWithLang = withLangContext(DropdownBodyEmpty, DropdownLang, sampleTexts);
  const DecoratedComponentWithLang = withLangContext(DropdownBodyEmptyDecorated, DropdownLang, sampleTexts);

  return {classes, classesKeys, emptyProps, sampleTexts, ComponentWithLang, DecoratedComponentWithLang};
};

describe('DropdownBodyEmpty component', () => {

  it('no options state match snapshot', () => {
    const { emptyProps, ComponentWithLang } = setup();
    const props = {
      ...emptyProps,
      isEmpty: true,
    };
    const component = mount(<ComponentWithLang {...props} />);
    expect(component.debug()).toMatchSnapshot();
  });

  it('no filtered state match snapshot', () => {
    const { emptyProps, ComponentWithLang } = setup();
    const props = {
      ...emptyProps,
      isFilteredEmpty: true,
    };
    const component = mount(<ComponentWithLang {...props} />);
    expect(component.debug()).toMatchSnapshot();
  });

  it('skips render is no empty required', () => {
    const { emptyProps, ComponentWithLang } = setup();
    const component = mount(<ComponentWithLang {...emptyProps} />);
    const componentContent = component.find('DropdownBodyEmpty').childAt(0);
    expect(componentContent.length).toBe(0);
  })
});

describe('DropdownBodyEmpty styles', () => {
  it('contains principle classes', () => {
    const { classesKeys } = setup();
    expect(dropdownBodyEmptyStyles).toBeDefined();
    classesKeys.forEach((keyValue) => {
      // @ts-ignore
      expect(dropdownBodyEmptyStyles[keyValue]).toBeDefined();
    });
  });
});

describe('DropdownBodyEmpty decorators', () => {

  it('provides styled classes from decorators', () => {
    const { emptyProps, DecoratedComponentWithLang, classesKeys } = setup();
    const component = mount(<DecoratedComponentWithLang {...emptyProps} />);
    // @ts-ignore
    const assignedClasses = component.find('DropdownBodyEmpty').props().classes;
    classesKeys.forEach(keyValue => {
      expect(assignedClasses[keyValue]).toBeDefined();
    });
  });
});
