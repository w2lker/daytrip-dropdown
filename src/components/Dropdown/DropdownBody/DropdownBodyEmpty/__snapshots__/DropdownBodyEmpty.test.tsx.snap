// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownBodyEmpty component no filtered state match snapshot 1`] = `
"<WithLangContext(undefined, undefined) isEmpty={false} isFilteredEmpty={true} classes={{...}}>
  <DropdownBodyEmpty isEmpty={false} isFilteredEmpty={true} classes={{...}}>
    <div className=\\"dropdownBodyEmpty-wrapper\\">
      <div className=\\"dropdownBodyEmpty-icon\\">
        <svg viewBox=\\"0 0 68 68\\">
          <g fill=\\"#B2C0C7\\">
            <path d=\\"M65.8363636,55.9454545 L48.5272727,40.0272727 C47.2909091,38.9454545 45.4363636,38.9454545 44.2,40.1818182 L43.2727273,41.1090909 L40.6454545,38.4818182 C44.3545455,34.3090909 46.3636364,28.9 46.3636364,23.1818182 C46.3636364,10.3545455 36.0090909,0 23.1818182,0 C10.3545455,0 0,10.3545455 0,23.1818182 C0,36.0090909 10.3545455,46.3636364 23.1818182,46.3636364 C27.2,46.3636364 31.2181818,45.2818182 34.7727273,43.2727273 C35.5454545,42.8090909 35.7,41.8818182 35.3909091,41.1090909 C34.9272727,40.3363636 34,40.1818182 33.2272727,40.4909091 C30.1363636,42.1909091 26.7363636,43.1181818 23.1818182,43.1181818 C12.0545455,43.2727273 3.09090909,34.3090909 3.09090909,23.1818182 C3.09090909,12.0545455 12.0545455,3.09090909 23.1818182,3.09090909 C34.3090909,3.09090909 43.2727273,12.0545455 43.2727273,23.1818182 C43.2727273,28.4363636 41.2636364,33.5363636 37.5545455,37.2454545 C37.2454545,37.5545455 37.0909091,37.8636364 37.0909091,38.3272727 L37.0909091,38.6363636 C37.0909091,39.1 37.2454545,39.4090909 37.5545455,39.7181818 L41.1090909,43.2727273 L40.1818182,44.2 C38.9454545,45.4363636 38.9454545,47.2909091 40.0272727,48.5272727 L55.9454545,65.8363636 C57.1818182,67.2272727 59.0363636,68 61.0454545,68 C64.9090909,68 68,64.9090909 68,61.0454545 C68,59.1909091 67.2272727,57.3363636 65.8363636,55.9454545 Z M61.0454545,64.9090909 C59.9636364,64.9090909 58.8818182,64.4454545 58.2636364,63.6727273 L42.3454545,46.3636364 L46.3636364,42.3454545 L63.6727273,58.2636364 C64.4454545,59.0363636 64.9090909,59.9636364 64.9090909,61.0454545 C64.9090909,63.2090909 63.2090909,64.9090909 61.0454545,64.9090909 Z\\" fillRule=\\"nonzero\\" />
            <path d=\\"M23,6 C13.5727273,6 6,13.5727273 6,23 C6,32.4272727 13.5727273,40 23,40 C32.4272727,40 40,32.4272727 40,23 C40,13.5727273 32.4272727,6 23,6 Z M23,36.9090909 C15.2727273,36.9090909 9.09090909,30.7272727 9.09090909,23 C9.09090909,15.2727273 15.2727273,9.09090909 23,9.09090909 C30.7272727,9.09090909 36.9090909,15.2727273 36.9090909,23 C36.9090909,30.7272727 30.7272727,36.9090909 23,36.9090909 Z\\" fillRule=\\"nonzero\\" />
            <path d=\\"M23.4337349,12 C17.1686747,12 12,17.1686747 12,23.4337349 C12,24.373494 12.626506,25 13.5662651,25 C14.5060241,25 15.1325301,24.373494 15.1325301,23.4337349 C15.1325301,18.8915663 18.8915663,15.1325301 23.4337349,15.1325301 C24.373494,15.1325301 25,14.5060241 25,13.5662651 C25,12.626506 24.373494,12 23.4337349,12 Z\\" />
          </g>
        </svg>
      </div>
      <h5 className=\\"dropdownBodyEmpty-title\\">
        noFiltered title
      </h5>
      <p className=\\"dropdownBodyEmpty-description\\">
        noFiltered description
      </p>
    </div>
  </DropdownBodyEmpty>
</WithLangContext(undefined, undefined)>"
`;

exports[`DropdownBodyEmpty component no options state match snapshot 1`] = `
"<WithLangContext(undefined, undefined) isEmpty={true} isFilteredEmpty={false} classes={{...}}>
  <DropdownBodyEmpty isEmpty={true} isFilteredEmpty={false} classes={{...}}>
    <div className=\\"dropdownBodyEmpty-wrapper\\">
      <div className=\\"dropdownBodyEmpty-icon\\">
        <svg viewBox=\\"0 0 68 68\\">
          <g fill=\\"#B2C0C7\\">
            <path d=\\"M65.8363636,55.9454545 L48.5272727,40.0272727 C47.2909091,38.9454545 45.4363636,38.9454545 44.2,40.1818182 L43.2727273,41.1090909 L40.6454545,38.4818182 C44.3545455,34.3090909 46.3636364,28.9 46.3636364,23.1818182 C46.3636364,10.3545455 36.0090909,0 23.1818182,0 C10.3545455,0 0,10.3545455 0,23.1818182 C0,36.0090909 10.3545455,46.3636364 23.1818182,46.3636364 C27.2,46.3636364 31.2181818,45.2818182 34.7727273,43.2727273 C35.5454545,42.8090909 35.7,41.8818182 35.3909091,41.1090909 C34.9272727,40.3363636 34,40.1818182 33.2272727,40.4909091 C30.1363636,42.1909091 26.7363636,43.1181818 23.1818182,43.1181818 C12.0545455,43.2727273 3.09090909,34.3090909 3.09090909,23.1818182 C3.09090909,12.0545455 12.0545455,3.09090909 23.1818182,3.09090909 C34.3090909,3.09090909 43.2727273,12.0545455 43.2727273,23.1818182 C43.2727273,28.4363636 41.2636364,33.5363636 37.5545455,37.2454545 C37.2454545,37.5545455 37.0909091,37.8636364 37.0909091,38.3272727 L37.0909091,38.6363636 C37.0909091,39.1 37.2454545,39.4090909 37.5545455,39.7181818 L41.1090909,43.2727273 L40.1818182,44.2 C38.9454545,45.4363636 38.9454545,47.2909091 40.0272727,48.5272727 L55.9454545,65.8363636 C57.1818182,67.2272727 59.0363636,68 61.0454545,68 C64.9090909,68 68,64.9090909 68,61.0454545 C68,59.1909091 67.2272727,57.3363636 65.8363636,55.9454545 Z M61.0454545,64.9090909 C59.9636364,64.9090909 58.8818182,64.4454545 58.2636364,63.6727273 L42.3454545,46.3636364 L46.3636364,42.3454545 L63.6727273,58.2636364 C64.4454545,59.0363636 64.9090909,59.9636364 64.9090909,61.0454545 C64.9090909,63.2090909 63.2090909,64.9090909 61.0454545,64.9090909 Z\\" fillRule=\\"nonzero\\" />
            <path d=\\"M23,6 C13.5727273,6 6,13.5727273 6,23 C6,32.4272727 13.5727273,40 23,40 C32.4272727,40 40,32.4272727 40,23 C40,13.5727273 32.4272727,6 23,6 Z M23,36.9090909 C15.2727273,36.9090909 9.09090909,30.7272727 9.09090909,23 C9.09090909,15.2727273 15.2727273,9.09090909 23,9.09090909 C30.7272727,9.09090909 36.9090909,15.2727273 36.9090909,23 C36.9090909,30.7272727 30.7272727,36.9090909 23,36.9090909 Z\\" fillRule=\\"nonzero\\" />
            <path d=\\"M23.4337349,12 C17.1686747,12 12,17.1686747 12,23.4337349 C12,24.373494 12.626506,25 13.5662651,25 C14.5060241,25 15.1325301,24.373494 15.1325301,23.4337349 C15.1325301,18.8915663 18.8915663,15.1325301 23.4337349,15.1325301 C24.373494,15.1325301 25,14.5060241 25,13.5662651 C25,12.626506 24.373494,12 23.4337349,12 Z\\" />
          </g>
        </svg>
      </div>
      <h5 className=\\"dropdownBodyEmpty-title\\">
        noOptions title
      </h5>
      <p className=\\"dropdownBodyEmpty-description\\">
        noOptions description
      </p>
    </div>
  </DropdownBodyEmpty>
</WithLangContext(undefined, undefined)>"
`;
